File: src/data/courses/dsa/trees/path-relationship-problems/all-root-to-leaf-paths/code/python/all_root_to_leaf_paths.py
# ============================================================
# All Root-to-Leaf Paths
# ============================================================
class Solution:
    def binaryTreePaths(self, root):
        result = []
        def dfs(node, path):
            path += str(node.val)
            if not node.left and not node.right:
                result.append(path); return
            if node.left: dfs(node.left, path + "->")
            if node.right: dfs(node.right, path + "->")
        if root: dfs(root, "")
        return result

File: src/data/courses/dsa/trees/path-relationship-problems/all-root-to-leaf-paths/code/java/AllRootToLeafPaths.java
// ============================================================
// All Root-to-Leaf Paths
// ============================================================
import java.util.*;
class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> result = new ArrayList<>();
        if (root != null) dfs(root, "", result);
        return result;
    }
    void dfs(TreeNode node, String path, List<String> result) {
        path += node.val;
        if (node.left == null && node.right == null) { result.add(path); return; }
        if (node.left != null) dfs(node.left, path + "->", result);
        if (node.right != null) dfs(node.right, path + "->", result);
    }
}
